ichnaea/data/schema.py

```diff
@@ -0,0 +1,376 @@
+    signal = DefaultNode(Integer(), missing=0, validator=Range(-150, -1))
+    ta = DefaultNode(Integer(), missing=0, validator=Range(0, 63))
+
+    def deserialize(self, data, default_radio=None):
+        if data:
+            if 'radio' in data:
+                if isinstance(data['radio'], basestring):
+                    data['radio'] = RADIO_TYPE.get(
+                        data['radio'], self.fields['radio'].missing)
+
+                # If a default radio was set,
+                # and we don't know, use it as fallback
+                if (self.is_missing(data, 'radio')
+                        and default_radio is not None):
+                    data['radio'] = default_radio
+
+                # If the cell id >= 65536 then it must be a umts tower
+                if (data.get('cid', 0) >= 65536
+                        and data['radio'] == RADIO_TYPE['gsm']):
+                    data['radio'] = RADIO_TYPE['umts']
+
+            else:
+                data['radio'] = default_radio
+
+            # Treat cid=65535 without a valid lac as an unspecified value
+            if (self.is_missing(data, 'lac')
+                    and data.get('cid', None) == 65535):
+                data['cid'] = self.fields['cid'].missing
+
+        return super(ValidCellBaseSchema, self).deserialize(data)
+
+    def validator(self, schema, data):
+        lac_missing = self.is_missing(data, 'lac')
+        cid_missing = self.is_missing(data, 'cid')
+
+        if data['mcc'] not in ALL_VALID_MCCS:
+            raise Invalid(
+                schema, 'Check against the list of all known valid mccs')
+
+        if (data['radio'] == RADIO_TYPE['cdma']
+                and (lac_missing or cid_missing)):
+            raise Invalid(schema, (
+                'Skip CDMA towers missing lac or cid '
+                '(no psc on CDMA exists to backfill using inference)'))
+
+        radio_types = RADIO_TYPE['gsm'], RADIO_TYPE['umts'], RADIO_TYPE['lte']
+        if data['radio'] in radio_types and data['mnc'] > 999:
+            raise Invalid(
+                schema, 'Skip GSM/LTE/UMTS towers with an invalid MNC')
+
+        if ((lac_missing or cid_missing) and self.is_missing(data, 'psc')):
+            raise Invalid(schema, (
+                'Must have (lac and cid) or '
+                'psc (psc-only to use in backfill)'))
+
+
+class ValidCellSchema(ValidCellBaseSchema):
+    """
+    A Schema which validates the fields present in
+    all Cells.
+    """
+    created = SchemaNode(DateTimeFromString(), missing=None)
+    modified = SchemaNode(DateTimeFromString(), missing=None)
+    changeable = SchemaNode(Boolean(), missing=True)
+    total_measures = SchemaNode(Integer(), missing=0)
+    range = SchemaNode(Integer(), missing=0)
+
+
+class ValidCellMeasureSchema(ValidCellBaseSchema):
+    """
+    A Schema which validates the fields present in
+    all Cell measurements.
+    """
+    created = SchemaNode(String(), missing=None)
```

Comment:
This mirrors the existing behaviour where created is a string and isn't validated or checked against.  If we want to change this behaviour it should be addressed in a separate ticket.